src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -154,4 +189,111 @@ private void ConfigureAspireDashboardResource(IResource dashboardResource)
+                    {
+                        dashboardResourceId = notification.ResourceId;
+                        break;
+                    }
+                }
+            }
+            catch (OperationCanceledException)
+            {
+            }
+
+            return dashboardResourceId;
+        }
+
+        if (await GetDashboardResourceIdentifier().ConfigureAwait(false) is not string dashboardResourceId)
+        {
+            // The only way this can happen is if DCP is unhealthy, so log a warning.
+            distributedApplicationLogger.LogWarning("Failed to get the dashboard logs.");
+            return;
+        }
+
+        var loggerCache = new Dictionary<string, ILogger>();
+        var defaultDashboardLogger = loggerFactory.CreateLogger("Aspire.Hosting.Dashboard");
+
+        try
+        {
+            // We turned on the JSON formatter for the logger, so we can log the log lines as JSON.
+            await foreach (var batch in resourceLoggerService.WatchAsync(dashboardResourceId).WithCancellation(_shutdownTcs.Token).ConfigureAwait(false))
+            {
+                foreach (var logLine in batch)
+                {
+                    try
+                    {
+                        var logMessage = JsonSerializer.Deserialize(logLine.Content, DashboardLogMessageContext.Default.DashboardLogMessage);
+
+                        if (logMessage is not null)
+                        {
+                            if (!loggerCache.TryGetValue(logMessage.Category, out var logger))
+                            {
+                                // Looks strange to see Aspire.Hosting.Dashboard.Aspire.Dashboard.Category,
+                                // so trim the prefix and append Aspire.Hosting.Why is this important?
+                                // Well there are logs emitting from categories that don't start with Aspire.Dashboard so we want to prefix all logs so that they can be controlled by config.
+                                var categoryTrimmed = logMessage.Category.StartsWith("Aspire.Dashboard.") ?
+                                    logMessage.Category["Aspire.Dashboard.".Length..] : logMessage.Category;
+
+                                loggerCache[logMessage.Category] = logger = loggerFactory.CreateLogger($"Aspire.Hosting.Dashboard.{categoryTrimmed}");
+                            }
+
+                            logger.Log(logMessage.LogLevel, logMessage.EventId, logMessage.Message, null, (s, _) => s);
+                        }
+                    }
+                    catch (JsonException)
+                    {
+                        // Failed to parse, it's not JSON, just log the content as is.
+                        var level = logLine.IsErrorMessage ? LogLevel.Error : LogLevel.Information;
+                        defaultDashboardLogger.Log(level, 0, logLine.Content, null, (s, _) => s);
```

Comment:
Should we log a warning here as well that the log output is malformed. Could be the sign of a stray Console.Write(...).

Reply:
I don't think so. If they are stray console.writelines we shouldn't warn.